New design system retain data

minhngocd opened the pull request:
The design system form did not persist data when failing to save. We needed to read the value from the document model. Also did some refactorings to make the facet field in the form a view component to better navigate. 

https://trello.com/c/5JnIybdU/3637-design-system-add-new-document-persist-form-content-on-failed-save-draft

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

minhngocd committed:
Retain data on new design system when failing to save

If there is an invalid field, retain data entered. This commit scopes it to only common fields

minhngocd committed:
Refactor: move specialist document facet input into a view component

minhngocd committed:
Split specific facet input components into their own view component

Specialist Document facets conditionally render a particular input view component depending on the facet type

minhngocd committed:
Facets to retain data when form fails to save

Reload the values from the document model.
The add another component loads the items, but the delete button doesn't seem to remove the items. It is only able to delete newly added items on the page. This bug can be descoped to be looked at in another ticket.

lauraghiorghisor-tw reviewed (approved):
LGTM

- I think the `input` bit in the names is a bit distracting, especially since input has other connotations in this space. I think `SpecialistDocumentFacetComponent` & co sound just fine. Could even drop the specialist document, if you ask me 🙈
- The textarea changes (value) can probably be moved from the last commit, as it seems to be more about multi and single select business, but not fussed.

minhngocd committed:
Rename facet input field components

Remove redundant "Input" as the sub components are all scoped under the namespace of "Facet Input Component".
Renamed SpecialistDocumentFacetInputComponent as Specialist Document is not necessary as part of the name
